sources/api/bootstrap-containers/src/main.rs

```diff
@@ -0,0 +1,722 @@
+/*!
+# Bootstrap containers
+
+Bootstrap containers are host containers that can be used to setup the host
+before services like the ecs-agent, kubelet and docker start.
+
+This type of containers are very similar to normal host containers, they come with
+persistent storage and with optional user data. Unlike normal host containers, bootstrap
+containers can't be treated as `superpowered`. However, these containers have access to
+the underlying root filesystem through `/.bottlerocket/rootfs`.
+
+Bootstrap containers have three different modes:
+
+* `always`: with this setting, the container is executed on every boot.
+* `off`: blocks the execution of the container
+* `once`: with this setting, the container only runs on the first boot. Upon completion,
+the container marks itself as `off`. If you need to execute the containers after the first
+boot, you can mark it as `once` and reboot.
+
+Bootstrap containers are executed in parallel during the execution of the configured target,
+without following a deterministic order. Therefore, bootstrap containers must be independent
+from each other. Bootstrap containers have the `essential` flag (which defaults to `false`),
+which can be used to fail the boot process if the underlying container task exits with a
+non-zero status code.
+
+Bootstrap containers are created in the `bootstrap` containerd's namespace to prevent
+collisions with other host containers, therefore you can have use the name for both container
+types.
```

Comment:
I thought on doing so but then I saw how tightly coupled is the `containerID` with other things like setting up the local mount filesystem (I had exactly the thing you asked for while I was implementing this change). If I prefix the container id with something within `host-ctr`, we need to extract what the "container name" is to setup the local persistent mount, otherwise existing customer images that already rely on `/.bottlerocket/host-containers/<container-name>` will be broken for the next release. 

These are the implementations that I tested (but didn't like them):

- pass `--container-id=<id> --container-type=<type>`, and let `host-ctr` figure out the correct prefix: I didn't like this idea because the caller is using a specific id and that id should be effectively used, but that is not the case because depending on the type we change the id and prefix it with something.

- pass `--container-id=<id> --container-type=<type>` where `--container-id` already contains the prefix: I like this idea better because now `host-ctr` effectively uses the passed container id. Still, within `host-ctr` there should be logic to get the correct container name to setup the local persistent mount.

Reply:
> pass `--container-id=<id> --container-type=<type>`, and let host-ctr figure out the correct prefix

This is what I would recommend.

> I didn't like this idea because the caller is using a specific id and that id should be effectively used, but that is not the case because depending on the type we change the id and prefix it with something.

I don't really think we've made that guarantee.  That's the way host-ctr behaves today, but the way we implement host containers is sufficiently an implementation detail that I'm not afraid of changing it; we've even (long ago) considered whether we wanted to eventually move away from containerd and toward something like systemd-nspawn.  As long as the ID you use to interact with host-ctr remains the same, the ID that is used with containerd isn't a guaranteed interface.

If we _do_ want to keep the existing host containers with the same ID (for example because we have reason to believe that someone has taken a dependency on them), we could reserve the prefixing only for bootstrap containers.